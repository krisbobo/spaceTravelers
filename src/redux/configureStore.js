import { configureStore } from '@reduxjs/toolkit';
import rocketReducer from './rockets/rockets';
import missionReducer from './missions/missions';
import dragonReducer from './dragons/dragons';

const store = configureStore({
  reducer: {
    rockets: rocketReducer,
    missions: missionReducer,
    dragons: dragonReducer,
  },
});
export default store;
