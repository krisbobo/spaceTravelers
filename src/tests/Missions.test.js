import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import Mission from '../components/MissionProfile';

const mockStore = configureStore([]);

describe('My connected React-Redux component', () => {
  let store;
  let component;

  beforeEach(() => {
    store = mockStore({
      missions: [
        {
          mission_name: 'Thaicom',
        },
      ],
    });

    component = renderer.create(
      <Provider store={store}>
        <Mission />
      </Provider>,
    );
  });
  it('should render with given state from Redux store', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
