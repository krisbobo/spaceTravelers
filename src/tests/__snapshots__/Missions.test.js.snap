// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My connected React-Redux component should render with given state from Redux store 1`] = `
<ul>
  <span>
     No Reserved Missions!
  </span>
</ul>
`;
